### DISCUSSION TITLE
SELECT FOR UPDATE - Avoid Mid-Air-Collisions with locking

### DISCUSSION THREAD

**Discussion by brandonmpetty** [prisma/prisma#8564] [Q&A] [answered]
Hey all,

I am using Prisma in my REST service and need to avoid Mid-Air-Collisions using an '**If-Unmodified-Since**' date value.
This seems perfect since the ORM has great **@updatedAt** support.

The problem is that I cannot figure out how to lock with Prisma.
Ideally I would do a **SELECT FOR UPDATE** lock for a specific record.  If I get back a result I would check its updatedAt value.  If it has been updated since my specified time, I would then return a 412 HTTP error.  If it is hasn't been updated, I would then issue an update.  If I can't find the record in the first place (delete row), I take other action.  Fini.

The problem is that I cannot figure out how to do this with Prisma.  Any assistance?

I tried a hacky work around with no luck either:
It appears that I can't filter by my primary key, id, and updatedAt using update().  The work around is using updateMany().  The problem is it returns a count, not my updated object.  I also can't tell if a failure to update was due to the updatedAt filter failing, or for some other reason, like a bad id, without some kind of intense string parsing.

**Comment by ryands17** [comment]
@brandonmpetty 👋 

The only way I can see this being done is creating a compound unique index like this:

```prisma
@@unique([id, updatedAt])
```

This will help you update the item directly via `update`.

**Reply by brandonmpetty** [reply]
Thanks @ryands17
Is there anyway to tell Prisma that id + updatedAt is Unique without it creating another index in the database?

**Reply by ryands17** [reply]
Nope. `@@unique` will always create an index as that's how it can know if there aren't any duplicates.

**Comment by brandonmpetty** [comment, marked as answer]
I have proposed a solution here: https://github.com/prisma/prisma/issues/8580

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
